results-processor/processor_test.py

```diff
@@ -232,3 +233,42 @@ def test_download_content_disposition(self):
             # artifact_test.zip as the filename.
             path = p._download_single(self.url + '/download/attachment')
             self.assertTrue(path.endswith('.zip'))
+
+
+class ProcessorAPIServerTest(unittest.TestCase):
+    def setUp(self):
+        self.server, self.url = test_util.start_server(True)
+
+    def tearDown(self):
+        if self.server.poll() is None:
```

Comment:
What does it mean for poll() to return None, or to return a non-None value? (I.e. I'm trying to understand when we will/won't call kill())

Reply:
`poll` returns the `returncode` of a terminated process. It is pretty much `wait(2)` with `WNOHANG`. If the process is still running, it returns `None`.

If you notice the `tearDown` methods are different in the two new classes, it is intentional. In this class, we capture the output of the test server process using `communicate`, which would block until the process terminates, so we call `terminate` in the tests before `communicate`. (There are other ways to read stdout/err, but this is the easiest way IMO for our use case.) This however creates the possibility of never calling `terminate` (if tests fail prematurely), so we have to double check and kill the server if needed in `tearDown`.